### ISSUE TITLE
Modern JS not transformed to ES3

### ISSUE THREAD

**Issue by Klustre** [hyperbrew/bolt-cep#11]
Hey @justintaylor-dev, I wanted to see how you did the ES6 support for Extendscript and noticed that ES6 array functions don't get ponyfilled. Haven't checked any other ES6 features.

Input
```ts
export const helloWorld = () => {
  const x = ['hello', 'from', 'ae']
  x.forEach((e) => alert(e))
  x.map((e) => `${e}gibberish`)
};
```
Output
```js
var r = function() {
  var e = [ "hello", "from", "ae" ];
  e.forEach(function(e) {
      return alert(e);
  });
  e.map(function(e) {
      return "".concat(e, "gibberish");
  });
};
```

**Comment by Klustre** [comment]
I've been using [babel-preset-extendscript](https://github.com/fusepilot/babel-preset-extendscript), but it only supports most ES2015 features. [core-js-pure](https://github.com/zloirock/core-js/blob/32492dc5f55ea9bbd21b8fa8145cedd36d68c160/README.md#babelruntime) has a ponyfill feature, but I haven't been able to have the bundler produce valid Extendscript (likely because of nested ternaries).

**Comment by Silly-V** [comment]
I have been just using some polyfill code and it works just fine, what you speak of is probably more advanced and proper.

**Comment by justintaylor-dev** [comment]
@Klustre Appreciate the suggestion! I've done some testing with these methods and I don't think I'll include these in the boilerplate for the time being. I have a util library for most common array methods in the [utils.ts](https://github.com/hyperbrew/bolt-cep/blob/master/src/jsx/utils/utils.ts) file.

If you want to add to your own clone, you can by installing and adding [babel-preset-extendscript](https://github.com/fusepilot/babel-preset-extendscript) to babel presets array in `vite.es.config.ts`, but you'll need to adjust the tsconfig to not show errors with modern array methods.

**Comment by justintaylor-dev** [comment]
@Klustre revisited this and added a simple implementation for Ponyfills in Rollup.

Built-in Ponyfills have been implemented in 0.4.0 and you can add your own custom ones (see the Readme for details).
